reach2

```diff
@@ -2,17 +2,20 @@
 
 IMG='reachsh/reach-cli:latest'
 
-if ! (which make docker docker-compose >/dev/null 2>&1) ; then
+run () {
+  docker run -t --rm "$IMG" "$@"
+}
+
+if ! (which make docker docker-compose >/dev/null 2>&1); then
   echo "Reach relies on an installation of make, docker, and docker-compose"
   exit 1
 fi
 
-if [ "$(docker image ls -q $IMG)" = '' ] ; then
-  docker pull "$IMG"
-  docker run -t --rm "$IMG" update | sed 's/\r//' | sh
+if [ "$(docker image ls -q $IMG)" = '' ]; then
+  if ! (docker pull "$IMG" && run update | sed 's/\r//' | sh); then exit 1; fi
```

Comment:
You get the exit for free if you do `/bin/sh -e`

Reply:
Oh, but I guess we can't because of the exit 42

Reply:
Yeah, for that and a few other reasons I don't think we can use `set -e` or `sh -e` in the wrapper script.

(The interpreted script coming from our Haskell code does `set -e` though.)